src/Microsoft.AspNetCore.WebHooks.Common/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel;
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// Extension methods for <see cref="IDictionary{TKey,TValue}"/>.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class DictionaryExtensions
```

Comment:
All these extension methods types need to be internal. We don't want them "leaking out" of this assembly.